### PR TITLE
Add more details to README

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR widens the `README.md` with more details on the project structure and links to other resources.

The objective of the resulting document is to get the newcomer developers a kick-start on the Spine development.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@meizabel, PTAL.
@alexander-yevsyukov, @armiol, PTAL as well.

File: README.md [hunk 1]
```diff
@@ -45,34 +54,39 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
- [Google Datastore](https://github.com/SpineEventEngine/gcloud-java) and 
+ * Configure the storage factory, that corresponds to the target environment. Connectors to
+ [Google Cloud Datastore](https://github.com/SpineEventEngine/gcloud-java) and 
  [JDBC-enabled storages](https://github.com/SpineEventEngine/jdbc-storage) are provided by Spine.
- * Append the creation of the bounded context with the registration of newly created repositories.
- 
- Other possible changes include dealing with security (e.g. authentication/authorization), defining
- deployment scheme, scaling approach etc. These and other advanced topics aren't covered by this sample.
- 
+ * Register the newly created repositories in the bounded context.
```
Comment (alexander-yevsyukov) [inline README.md]: Please capitalize Bounded Context.

File: README.md [hunk 2]
```diff
@@ -15,18 +17,25 @@ The project consists of three modules.
 ### The `model` Module
 
 Defines the [ubiquitous language](https://martinfowler.com/bliki/UbiquitousLanguage.html)
```
Comment (alexander-yevsyukov) [inline README.md]: Please capitalize Ubiquitous Language.

File: README.md [hunk 3]
```diff
@@ -15,18 +17,25 @@ The project consists of three modules.
 ### The `model` Module
 
 Defines the [ubiquitous language](https://martinfowler.com/bliki/UbiquitousLanguage.html) 
-of the application in Protobuf. See `model/proto` folder.
+of the application in Protobuf.
+The `model/src/main/proto` directory contains the Protobuf definitions of the domain model:
+ * `Task` is an aggregate state type; as any entity type, it is marked with the `(entity)` option;
+ * `TaskCreated` in `events.proto` is an event of the `Task` aggregate;
```
Comment (alexander-yevsyukov) [inline README.md]: `TaskAggregate` must be a class name. Otherwise it is confusing.

File: README.md [hunk 4]
```diff
@@ -45,34 +54,39 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
-and demonstrates a really simple RPC interaction. To take it closer to the production needs, 
+and demonstrates a really simple RPC interaction. To take it closer to the production needs,
 the following steps are suggested:
 
 ### Gradle Configuration Changes:
 
  * Update `*.gradle` files with the artifact attributes, that correspond to your project.
 
-### Suggested `model` Changes 
+### Suggested `model` Changes
```
Comment (alexander-yevsyukov) [inline README.md]: There must be a simple change to a model. Without surrounding waters. Add a field to a command, add one more command. Make it really simple.

File: README.md [hunk 5]
```diff
@@ -45,34 +54,39 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
- * Together with the domain experts, perform event storming in the bounded context of the domain 
+ * Together with the domain experts, perform event storming in the bounded context of the domain
  speaking in language, ubiquitous for this domain.
- * According to the event storming results, define events, commands, entity states and value 
-objects in Protobuf. Put them into the `model/proto` folder. 
- * Design and implement Aggregates, Projections and Process Managers on top of the previously 
- defined language elements. Create repositories for them.
+ * According to the event storming results, define events, commands, entity states, and value
+objects in Protobuf. 
+ * Design and implement Aggregates, Projections and Process Managers on top of the previously
```
Comment (alexander-yevsyukov) [inline README.md]: Remove water.

**Review by alexander-yevsyukov — CHANGES_REQUESTED** [review]
Please simplify the text as we discussed vocally recently. Remove all abstract things. Do not make it a tutorial. For changing current code suggest something simple, like adding a field to a command.

File: README.md [hunk 6]
```diff
@@ -44,35 +53,87 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
-For more information please visit [the official website](https://spine.io).
+ * Experiment with the model—create a new command type:
+```proto
+// -- commands.proto --
+
+message AssignDueDate {
+    
+    TaskId task_id = 1;
+    
+    google.protobuf.Timestamp due_date = 2 [(when).in = FUTURE];
```
Comment (alexander-yevsyukov) [inline README.md]: Please use `LocalDate` instead.

File: README.md [hunk 7]
```diff
@@ -44,35 +53,87 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
- Other possible changes include dealing with security (e.g. authentication/authorization), defining
- deployment scheme, scaling approach etc. These and other advanced topics aren't covered by this sample.
- 
-
-Typically these steps are repeated for each bounded context in the application.
- 
-For more information please visit [the official website](https://spine.io).
+ * Experiment with the model—create a new command type:
+```proto
+// -- commands.proto --
```
Comment (alexander-yevsyukov) [inline README.md]: Have something like: Please add the following message to `commands.proto` file.

File: README.md [hunk 8]
```diff
@@ -44,35 +53,87 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
+message AssignDueDate {
+    
+    TaskId task_id = 1;
+    
+    google.protobuf.Timestamp due_date = 2 [(when).in = FUTURE];
+}
+```
+and a new event type:
+```proto
+// -- events.proto --
```
Comment (alexander-yevsyukov) [inline README.md]: Similarly, please add instruction for updating `events.proto` file.

File: README.md [hunk 9]
```diff
@@ -44,35 +53,87 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
+
+
+    // A title of the task.
+    string title = 2 [(required) = true];
+    
+    // The date and time by which this task should be completed.
+    google.protobuf.Timestamp due_date = 3; // <-- New field
+}
+```
+Make sure to run a Gradle build after the changing the Protobuf definitions.
```
Comment (alexander-yevsyukov) [inline README.md]: Please add command line.

File: README.md [hunk 10]
```diff
@@ -44,35 +53,87 @@ Run `ClientApp.main()` to start the client and see it connecting to the server.
...
+}
+```
+ * Apply the emitted event:
+```java
+@Apply
+private void on(DueDateAssigned event) {
+    builder().setDueDate(event.getDueDate());
+}
+```
+ * In `ClientApp`, append the `main()` method. Post another command:
```
Comment (alexander-yevsyukov) [inline README.md]: Did you mean ”extend”? The word ”append” sounds as we're adding yet another method to the bottom of the class source code.

**Review by alexander-yevsyukov — CHANGES_REQUESTED** [review]
We're close. Please see my comments.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL again.

File: client/src/main/java/io/spine/quickstart/client/ClientApp.java [hunk 11]
```diff
@@ -75,29 +77,33 @@ private ClientApp() {
      *
      * <p>Uses the hard-coded {@linkplain #HOST host} and {@linkplain #PORT port} for simplicity.
      */
-    public static void main(String[] args) throws InterruptedException {
+    public static void main(String[] args) throws InterruptedException, ParseException {
```
Comment (alexander-yevsyukov) [inline client/src/main/java/io/spine/quickstart/client/ClientApp.java]: Do we still need `ParseException`?

File: README.md [hunk 12]
```diff
@@ -41,38 +50,92 @@ Interacts with the gRPC services, exposed by the `server` module:
...
+Remember to import `LocalDate` via `import "spine/time/time.proto";`. This type is provided by
+the [Spine Time](https://github.com/SpineEventEngine/time) library. You don't have to perform any
+additional steps to use it in your domain.
+ * Create a new event type in `events.proto`:
+```proto
+message DueDateAssigned {
+
+    TaskId task_id = 1;
+
+    spine.time.LocalDate due_date = 2;
```
Comment (alexander-yevsyukov) [inline README.md]: Isn't it `required`?

File: README.md [hunk 13]
```diff
@@ -41,38 +50,92 @@ Interacts with the gRPC services, exposed by the `server` module:
...
-Typically these steps are repeated for each bounded context in the application.
- 
-For more information please visit [the official website](https://spine.io).
+ * Experiment with the model. Create a new command type in `commands.proto`
+```proto
+message AssignDueDate {
+    
+    TaskId task_id = 1;
+    
+    spine.time.LocalDate due_date = 2;
```
Comment (alexander-yevsyukov) [inline README.md]: Isn't this `required`?

File: README.md [hunk 14]
```diff
@@ -41,38 +50,92 @@ Interacts with the gRPC services, exposed by the `server` module:
...
+}
+```
+ * Adjust the aggregate state:
+```proto
+message Task {
+    option (entity).kind = AGGREGATE;
+
+    // An ID of the task.
+    TaskId id = 1;
+
```
Comment (alexander-yevsyukov) [inline README.md:80]: Please remove extra line.

**Review by alexander-yevsyukov — COMMENTED** [review]
Please see my comments.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL again.

File: README.md [hunk 15]
```diff
@@ -41,38 +50,91 @@ Interacts with the gRPC services, exposed by the `server` module:
...
+        .newBuilder()
+        .setTaskId(taskId)
+        .setDueDate(LocalDates.of(2038, JANUARY, 19))
+        .build();
+commandService.post(requestFactory.command()
+                                  .create(dueDateCommand));
+```
+and log the updated state:
+```java
+QueryResponse updatedStateResponse = queryService.read(readAllTasks);
```
Comment (alexander-yevsyukov) [inline README.md]: Please read only the task we're dealing with.

File: client/src/main/java/io/spine/quickstart/client/ClientApp.java [hunk 16]
```diff
@@ -83,21 +85,25 @@ public static void main(String[] args) throws InterruptedException {
...
         ActorRequestFactory requestFactory = ActorRequestFactory
                 .newBuilder()
                 .setActor(whoIsCalling())
                 .build();
-        CreateTask createTask = newCreateTaskCommand("Wash my car");
+        TaskId taskId = TaskIdVBuilder
+                .newBuilder()
+                .setValue(newUuid())
+                .build();
+        CreateTask createTask = newCreateTaskCommand(taskId, "Wash my car");
```
Comment (alexander-yevsyukov) [inline client/src/main/java/io/spine/quickstart/client/ClientApp.java]: Please think of something else for the task.

**Review by alexander-yevsyukov — CHANGES_REQUESTED** [review]
Please see my comments.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**